diff --git a/src/meson.build b/src/meson.build
index eb316b0..be2f0f1 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -4,7 +4,6 @@ cargo_bin  = find_program('cargo')
 assert(cargo_bin.found())
 cargo_opt  = [ '--manifest-path', meson.project_source_root() / 'Cargo.toml' ]
 cargo_opt += [ '--target-dir', meson.project_build_root()  / 'src' ]
-cargo_env  = [ 'CARGO_HOME=' + meson.project_build_root()  / 'cargo-home' ]
 
 if get_option('buildtype') == 'release'
   cargo_opt += [ '--release' ]
@@ -32,7 +31,6 @@ custom_target(
   install: true,
   install_dir: join_paths(get_option('prefix'), get_option('bindir')),
   command: [
-    'env', cargo_env,
     cargo_bin, 'build', cargo_opt, '&&',
     'cp', '-f', binaries_path, '@OUTDIR@'
   ]
