--- mixxx-2.1.0/build/depends.py	2018-03-26 21:20:09.160438959 +0200
+++ mixxx-2.1.0/build/depends.py	2018-03-26 13:07:19.614659087 +0200
@@ -286,7 +286,6 @@
             elif not qt5 and not conf.CheckForPKG('QtCore', '4.6'):
                 raise Exception('QT >= 4.6 not found')
 
-            qt_modules.extend(['QtDBus'])
             # This automatically converts QtXXX to Qt5XXX where appropriate.
             if qt5:
                 build.env.EnableQt5Modules(qt_modules, debug=False)
--- mixxx-2.1.0/src/util/screensaver.cpp	2018-03-26 21:16:39.680346842 +0200
+++ mixxx-2.1.0/src/util/screensaver.cpp	2018-03-26 21:17:36.967637297 +0200
@@ -24,7 +24,6 @@
 #elif defined(Q_OS_WIN)
 #  include <windows.h>
 #elif defined(Q_OS_LINUX)
-#  include <QtDBus>
 #elif HAVE_XSCREENSAVER_SUSPEND
 #  include <X11/extensions/scrnsaver.h>
 #endif // Q_OS_WIN
@@ -224,54 +223,11 @@
 
 void ScreenSaverHelper::inhibitInternal()
 {
-    if (!QDBusConnection::sessionBus().isConnected()) {
-        qWarning("Cannot connect to the D-Bus session bus.\nTo start it, run:\n"
-                "\teval `dbus-launch --auto-syntax`");
-        return;
-    }
-    if (s_cookie > 0) {
-        uninhibit();
-    }
-    s_cookie = 0;
-    for (int i=0; SCREENSAVERS[i][0] != nullptr; i++ ) {
-        QDBusInterface iface(SCREENSAVERS[i][0], SCREENSAVERS[i][1], SCREENSAVERS[i][2], 
-            QDBusConnection::sessionBus());
-        if (iface.isValid()) {
-            QDBusReply<uint32_t> reply = iface.call(SCREENSAVERS[i][3], "org.mixxxdj","Mixxx active");
-            if (reply.isValid()) {
-                s_cookie = reply.value();
-                s_saverindex = i;
-                s_enabled = true;
-                qDebug() << "DBus screensaver " << SCREENSAVERS[i][0] <<" inhibited";
-                break;
-            } else {
-                qWarning() << "Call to inhibit for " << SCREENSAVERS[i][0] << " failed: " 
-                    << reply.error().message();
-            }
-        } else {
-            qDebug() << "DBus interface " << SCREENSAVERS[i][0] << " not valid";
-        }
-    }
+	return;
 }
 void ScreenSaverHelper::uninhibitInternal()
 {
-    if (s_cookie > 0) {
-        s_enabled = false;
-        QDBusInterface iface(SCREENSAVERS[s_saverindex][0], SCREENSAVERS[s_saverindex][1], 
-            SCREENSAVERS[s_saverindex][2],  QDBusConnection::sessionBus());
-        if (iface.isValid()) {
-            QDBusReply<void> reply = iface.call("UnInhibit", s_cookie);
-            if (reply.isValid()) {
-                s_cookie = 0;
-                qDebug() << "DBus screensaver " << SCREENSAVERS[s_saverindex][0] << " uninhibited";
-            } else {
-                qWarning() << "Call to uninhibit for " << SCREENSAVERS[s_saverindex][0] << " failed: " 
-                    << reply.error().message();
-            }
-        } else {
-            qDebug() << "DBus interface " << SCREENSAVERS[s_saverindex][0] << " not valid";
-        }
-    }
+	return;
 }
 
 #elif HAS_XWINDOW_SCREENSAVER
