--- src/Makefile	2013-11-10 00:12:11.000000000 +0100
+++ src/Makefile	2017-07-27 20:23:13.255683357 +0200
@@ -1,29 +1,66 @@
-CXXFLAGS= -O3 -fomit-frame-pointer
-CXX=clang++	# Use clang++, as g++ optimizations cause crashes...
+CXXFLAGS ?= -O2 -fomit-frame-pointer
 override CXXFLAGS+= -Wall -fsigned-char
 
 PLATFORM= $(shell uname -s)
 PLATFORM_PREFIX=native
 
-INCLUDES= -I. -Ibot -I../enet/include
+PKG_CONFIG ?= pkg-config
 
-STRIP=
-ifeq (,$(findstring -g,$(CXXFLAGS)))
-ifeq (,$(findstring -pg,$(CXXFLAGS)))
-  STRIP=strip
-endif
+INCLUDES= -I. -Ibot
+
+BUNDLED_ENET ?= YES
+ifeq ($(BUNDLED_ENET),YES)
+INCLUDES += -I../enet/include
+else
+INCLUDES += $(shell $(PKG_CONFIG) --cflags libenet)
 endif
 
+STRIP=
+
 MV=mv
 ifneq (,$(findstring MINGW,$(PLATFORM)))
+MINGWBUILD=yes
+endif
+
+ifeq ($(MINGWBUILD),YES)
 WINDRES= windres
 CLIENT_INCLUDES= $(INCLUDES) -I../include
 CLIENT_LIBS= -mwindows -L../lib -lSDL -lSDL_image -lzdll -lopengl32 -lenet -lOpenAL32 -llibvorbisfile -llibintl -lws2_32 -lwinmm -lcurl
 else
-USRLIB=$(shell if [ -e /usr/lib64 ]; then echo "/usr/lib64"; else echo "/usr/lib"; fi)
-# override CXXFLAGS+= -rdynamic		# clang++ doesn't use this...
-CLIENT_INCLUDES= $(INCLUDES) -I/usr/include `sdl-config --cflags` -idirafter ../include
-CLIENT_LIBS= -L../enet/.libs -lenet -L$(USRLIB) -lX11 `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile -lcurl
+override CXXFLAGS+= -rdynamic
+CLIENT_INCLUDES= \
+		 $(INCLUDES) \
+		 -I/usr/include \
+		 `sdl-config --cflags` \
+		 -idirafter ../include \
+		 $(shell $(PKG_CONFIG) --cflags SDL_image) \
+		 $(shell $(PKG_CONFIG) --cflags zlib) \
+		 $(shell $(PKG_CONFIG) --cflags gl) \
+		 $(shell $(PKG_CONFIG) --cflags openal) \
+		 $(shell $(PKG_CONFIG) --cflags vorbisfile) \
+		 $(shell $(PKG_CONFIG) --cflags x11)
+		 $(shell $(PKG_CONFIG) --cflags curl)
+
+ifneq ($(BUNDLED_ENET),YES)
+CLIENT_INCLUDES += $(shell $(PKG_CONFIG) --cflags libenet)
+endif
+
+CLIENT_LIBS= \
+	     `sdl-config --libs` \
+	     $(shell $(PKG_CONFIG) --libs SDL_image) \
+	     $(shell $(PKG_CONFIG) --libs zlib) \
+	     $(shell $(PKG_CONFIG) --libs gl) \
+	     $(shell $(PKG_CONFIG) --libs openal) \
+	     $(shell $(PKG_CONFIG) --libs vorbisfile) \
+	     $(shell $(PKG_CONFIG) --libs x11)
+		 $(shell $(PKG_CONFIG) --cflags curl)
+
+ifeq ($(BUNDLED_ENET),YES)
+CLIENT_LIBS += -L../enet/.libs -lenet
+else
+CLIENT_LIBS += $(shell $(PKG_CONFIG) --libs libenet)
+endif
+
 endif
 
 CLIENT_OBJS= \
@@ -83,12 +120,21 @@
 
 CLIENT_PCH= cube.h.gch
 
-ifneq (,$(findstring MINGW,$(PLATFORM)))
+ifeq ($(MINGWBUILD),YES)
 SERVER_INCLUDES= -DSTANDALONE $(INCLUDES) -I../include
 SERVER_LIBS= -L../lib -lzdll -lenet -llibintl -lws2_32 -lwinmm
 else
-SERVER_INCLUDES= -DSTANDALONE $(INCLUDES)
-SERVER_LIBS= -L../enet/.libs -lenet -lz
+SERVER_INCLUDES= \
+		 -DSTANDALONE \
+		 $(INCLUDES) \
+		 $(shell $(PKG_CONFIG) --cflags zlib)
+
+SERVER_LIBS= $(shell $(PKG_CONFIG) --libs zlib)
+ifeq ($(BUNDLED_ENET),YES)
+SERVER_LIBS += -L../enet/.libs -lenet
+else
+SERVER_LIBS += $(shell $(PKG_CONFIG) --libs libenet)
+endif
 endif
 
 SERVER_OBJS= \
@@ -155,13 +201,24 @@
 server_install: server
 
 else
+ifeq ($(BUNDLED_ENET),YES)
 client: libenet $(CLIENT_OBJS)
-	$(CXX) $(CXXFLAGS) -o ac_client $(CLIENT_OBJS) $(CLIENT_LIBS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_client $(CLIENT_OBJS) $(CLIENT_LIBS)
 
 server: libenet $(SERVER_OBJS)
-	$(CXX) $(CXXFLAGS) -o ac_server $(SERVER_OBJS) $(SERVER_LIBS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_server $(SERVER_OBJS) $(SERVER_LIBS)
 master: libenet $(MASTER_OBJS)
-	$(CXX) $(CXXFLAGS) -o ac_master $(MASTER_OBJS) $(SERVER_LIBS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_master $(MASTER_OBJS) $(SERVER_LIBS)
+else
+client: $(CLIENT_OBJS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_client $(CLIENT_OBJS) $(CLIENT_LIBS)
+
+server: $(SERVER_OBJS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_server $(SERVER_OBJS) $(SERVER_LIBS)
+
+master: $(MASTER_OBJS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_master $(MASTER_OBJS) $(SERVER_LIBS)
+endif
 
 client_install: client
 	install -d ../../bin_unix/
